src/Microsoft.AspNetCore.Razor.Design/Internal/AssemblyTagHelperDescriptorResolver.cs

```diff
@@ -70,5 +102,30 @@ protected virtual IEnumerable<Type> GetTagHelperTypes(string assemblyName, Error
         {
             return _tagHelperTypeResolver.Resolve(assemblyName, SourceLocation.Zero, errorSink);
         }
+
+        private void LoadMvcAssembly()
+        {
+            _loadedAssembly = true;
+
+            try
+            {
+                var assembly = Assembly.Load(new AssemblyName("Microsoft.AspNetCore.Mvc"));
+
+                var providerClass = assembly.GetType(
+                    "Microsoft.AspNetCore.Mvc.DesignTimeMvcServiceCollectionProvider",
+                    throwOnError: true);
+
+                _serviceCollectionProviderClassInstance = Activator.CreateInstance(providerClass);
+                _populateMethod = providerClass.GetMethod(
```

Comment:
Can we do something similar to this - https://github.com/aspnet/HttpAbstractions/blob/e3207557342f514deea4618f595a049bfbd7a024/src/Microsoft.AspNetCore.Http.Extensions/HeaderDictionaryTypeExtensions.cs#L223-L234 - to ensure we're getting the right overload of `PopulateServiceCollection`